### PR TITLE
Update dependencies, refactor code

### PR DESCRIPTION
**Pull request by sacOO7**
- Updated dependencies as per new releases for https://github.com/ably/laravel-broadcaster/pull/22 and https://github.com/ably-forks/laravel-echo/pull/18
- Locked dependencies to make sure the app will always work for fresh installation on new devices.
- Added `Sqlite` as a default DB to make sure the app setup is simple.
- Refactored code, and added doc comments to make it more readable.
- Completed tasks as per #8

### CODE REVIEW

File: resources/js/components/ChatComponent.vue [hunk 1]
```diff
@@ -108,7 +108,7 @@ export default {
     },
 
     methods: {
-        joinPublic(event) {
+        subscribeToPubliChannel(event) {
```
Comment (qsdstefan) [inline resources/js/components/ChatComponent.vue:111]: Typo here

Comment (sacOO7) [reply resources/js/components/ChatComponent.vue:111]: I didn't get the typo 🤔

**Review by qsdstefan — APPROVED** [review]
LGTM!

**Review by qsdstefan — APPROVED** [review]

**Review by mikelee638 — COMMENTED** [review]
Generally looks good. Left a few comments.

File: resources/js/components/ChatComponent.vue [hunk 2]
```diff
@@ -128,6 +128,7 @@ export default {
                 }
                 channelName = channelName.trim();
 
+                // Register and subsribe to events on the public channel.
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply resources/js/components/ChatComponent.vue]: updated!

File: resources/js/components/ChatComponent.vue [hunk 3]
```diff
@@ -310,7 +313,8 @@ export default {
 
         },
 
-        sendMessage(event) {
+        // Uses to send message to other clients using client-events.
+        publishMessageOnPrivateCahnnel(event) {
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply resources/js/components/ChatComponent.vue]: updated!

File: resources/js/components/ChatComponent.vue [hunk 4]
```diff
@@ -348,19 +352,21 @@ export default {
...
 
-            const broadcastUrl = window.location.origin + "/public-event";
-            axios.get(broadcastUrl, { params: { message, channel}});
+            const broadcastUrl = window.location.origin + "/api/public-event";
+            axios.post(broadcastUrl, { channelName, message });
 
             this.message = null;
         },
 
+        // Unsubsribe and leaves the channel
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply resources/js/components/ChatComponent.vue]: updated!

File: routes/api.php [hunk 5]
```diff
@@ -17,3 +19,36 @@
...
+// If using reverse proxy, make sure to configure IP properly, since all incoming requests will have same IP address.
+// https://dev.to/aliadhillon/new-simple-way-of-creating-custom-rate-limiters-in-laravel-8-65n.
+Route::middleware('throttle:60,1')->post('/public-event', function (Request $request) {
+    $channelName = $request->post('channelName');
+    $message = $request->post('message');
+    broadcast(new PublicMessageEvent( $channelName, $message ));
+});
+
+// Private broadcast for an authenticated user.
+// If throttling is enabled, it will be per user session insetad of ip adddress.
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply routes/api.php]: updated!

**Review by mikelee638 — APPROVED** [review]

**Pull request merged**